### PR TITLE
feat(console): add-saml-to-idp (#6687)

### LINKED ISSUE #6687: SAML SP Console
As an administrator I want to be able to administrate a SAML Idp from the console. The backend and APIs have already been implemented with the following issue: https://github.com/zitadel/zitadel/issues/2947

**Acceptance Criteria**
- [ ] I am able to choose SAML in the Identity Provider Section of the console
- [ ] I am able to fill the following SAML information: name, metadataXML, metadataUrl, binding, withSignedRequest
- [ ] I am able to either fill the URL to a SAML metadata file or a uploading a metadata xml (see screenshot in the comment below)
- [ ] General Provider Informations can be filled: isLinkingAllowed, isCreationAllowed, isAutoCreation, isAutoUpdate

** Additional Information**
- Add SAML IDP API: https://zitadel.com/docs/apis/resources/mgmt/management-service-add-saml-provider
- Update SAML IDP API: https://zitadel.com/docs/apis/resources/mgmt/management-service-update-saml-provider

### PR DESCRIPTION
**Pull request by misko626**
Added functionality for selecting SAML Identity provider option in the console app. 

closes #6687

### Definition of Ready

- [x] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by eliobischof** [comment]
@stebenz should the SAML SP template be returned by the ListProviders endpoint since https://github.com/zitadel/zitadel/pull/6454?

Because it is not mapped in the backend response and that breaks the console in this PR
image

**Comment by misko626** [comment]
Hi @eliobischof. So should just the biding be a string, not the SAMLBiding enum value?

**Comment by eliobischof** [comment]
> Hi @eliobischof. So should just the biding be a string, not the SAMLBiding enum value?

Never mind, the enum is ok like you used it.
it's a backend issue that's already fixed in the other PR that will probably be merged today. I'll test and review again after that.

File: console/src/assets/i18n/en.json [hunk 1]
```diff
@@ -1837,6 +1841,12 @@
       "UPLOADPRIVATEKEY": "Upload Private Key",
       "KEYMAXSIZEEXCEEDED": "Maximum size of 5kB exceeded."
     },
+    "SAML": {
+      "METADATAXML": "Metadata Xml",
+      "METADATAURL": "Metadata URL",
+      "BINDING": "Biding",
```
Comment (eliobischof) [inline console/src/assets/i18n/en.json:1847]:
```suggestion
      "BINDING": "Binding",
```

File: console/src/assets/i18n/en.json [hunk 2]
```diff
@@ -1837,6 +1841,12 @@
       "UPLOADPRIVATEKEY": "Upload Private Key",
       "KEYMAXSIZEEXCEEDED": "Maximum size of 5kB exceeded."
     },
+    "SAML": {
+      "METADATAXML": "Metadata Xml",
+      "METADATAURL": "Metadata URL",
+      "BINDING": "Binding",
+      "SIGNEDREQUEST": "Signed Request"
+    },
```
Comment (eliobischof) [inline console/src/assets/i18n/en.json:1849]: Please translate to other languages or give us access to edit the PR :slightly_smiling_face:

**Comment by peintnermax** [comment]
@misko626 we need an additional change in the `idp-table.component.html` which shows the table entry for saml

```
 <div class="idp-table-provider-type" *ngSwitchCase="ProviderType.PROVIDER_TYPE_SAML">
              <img class="idp-logo" src="./assets/images/idp/saml-icon.svg" alt="saml" />
              SAML SP
            </div>
```

the rest looks great to me 👍

**Comment by eliobischof** [comment]
> @misko626 we need an additional change in the `idp-table.component.html` which shows the table entry for saml
> 
> ```
>  <div class="idp-table-provider-type" *ngSwitchCase="ProviderType.PROVIDER_TYPE_SAML">
>               saml
>               SAML SP
>             </div>
> ```
> 
> the rest looks great to me 👍 

@peintnermax the translations are still missing, right?

**Comment by misko626** [comment]
@eliobischof I have not added translations yet.

**Comment by peintnermax** [comment]
> > @misko626 we need an additional change in the `idp-table.component.html` which shows the table entry for saml
> > ```
> >  <div class="idp-table-provider-type" *ngSwitchCase="ProviderType.PROVIDER_TYPE_SAML">
> >               saml
> >               SAML SP
> >             </div>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the rest looks great to me 👍
> 
> @peintnermax the translations are still missing, right?

The translations of the saml fields are missing but we dont translate the technologies as they stay the same on all languages 👍

**Comment by misko626** [comment]
Can someone please approve this?

**Review by peintnermax — APPROVED** [review]

**Pull request merged**